hack/make-rules/test-cmd-util.sh

```diff
@@ -1083,6 +1032,137 @@ run_kubectl_apply_tests() {
+  kubectl delete pods test-pod-2 "${kube_flags[@]}"
+
+  ## If different label specified, then pod does not get pruned
+  # First, ensure that no pods exist
+  kube::test::get_object_assert pods "{{range.items}}{{$id_field}}:{{end}}" ''
+  # Next, create pod with apply, and ensure it exists
+  kubectl apply -f hack/testdata/pod.yaml "${kube_flags[@]}"
+  kube::test::get_object_assert 'pods test-pod' "{{${labels_field}.name}}" 'test-pod-label'
+  # Now, prune specifying new pods label, which does not prune previously created pod
+  kubectl apply --prune -l name=test-pod-label-2 -f hack/testdata/pod-2.yaml "${kube_flags[@]}"
+  kube::test::get_object_assert 'pods test-pod' "{{${labels_field}.name}}" 'test-pod-label'
+  kube::test::get_object_assert 'pods test-pod-2' "{{${labels_field}.name}}" 'test-pod-label-2'
+  # Clean up pods
+  kubectl delete pods test-pod "${kube_flags[@]}"
+  kubectl delete pods test-pod-2 "${kube_flags[@]}"
+
+  ## Test prune and dryRun does not really prune
+  # First, ensure that no pods exist
+  kube::test::get_object_assert pods "{{range.items}}{{$id_field}}:{{end}}" ''
+  # Next, create pod with apply, and ensure it exists
+  kubectl apply -f hack/testdata/pod.yaml "${kube_flags[@]}"
+  kube::test::get_object_assert 'pods test-pod' "{{${labels_field}.name}}" 'test-pod-label'
+  # Now, prune with dry-run does not actually create test-pod-2 or prune test-pod
+  kubectl apply --dry-run --prune --all -f hack/testdata/pod-2.yaml "${kube_flags[@]}"
+  kube::test::get_object_assert 'pods test-pod' "{{${labels_field}.name}}" 'test-pod-label'
+  output_message=$(! kubectl get pods test-pod-2 2>&1 "${kube_flags[@]}")
+  kube::test::if_has_string "${output_message}" 'pods "test-pod-2" not found'
+  # Clean up pods
+  kubectl delete pods test-pod "${kube_flags[@]}"
+
+  ## Non-namespaced resources get pruned (PersistentVolume) example
+  # First, ensure that no persistentvolumes exist
+  kube::test::get_object_assert persistentvolumes "{{range.items}}{{$id_field}}:{{end}}" ''
+  # Next, create persistentvolume through apply, and ensure it exists
+  kubectl apply -f hack/testdata/pv.yaml "${kube_flags[@]}"
+  kube::test::get_object_assert persistentvolumes "{{range.items}}{{$id_field}}:{{end}}" 'pv-test:'
+  # Next, apply a new peristentvolume and check the previous one was pruned
+  kubectl apply --prune --all -f hack/testdata/pv-2.yaml "${kube_flags[@]}"
+  kube::test::get_object_assert persistentvolumes "{{range.items}}{{$id_field}}:{{end}}" 'pv-test-2:'
+  kubectl delete persistentvolumes pv-test-2
+  
+  ## Test pruning only affects resources applied in "visited" namespaces.
+  kube::test::get_object_assert pods "{{range.items}}{{$id_field}}:{{end}}" ''
+  kubectl apply -f hack/testdata/pod.yaml
+  kube::test::get_object_assert 'pods test-pod' "{{${labels_field}.name}}" 'test-pod-label'
+  # Create pod in current namespace; pruning should not delete first pod
+  current_context=$(kubectl config current-context)
```

Comment:
> current_context

Is this used anywhere?

Reply:
This is copied verbatim from a previous test. Let's keep it the same in this PR, and we'll do fixes in following PR's.